packages/eslint-plugin/src/rules/class-literals-style.ts

```diff
@@ -0,0 +1,141 @@
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      preferFieldStyle: 'Literals should be exposed using readonly fields.',
+      preferGetterStyle: 'Literals should be exposed using getters.',
+    },
+    schema: [{ enum: ['fields', 'getters'] }],
+  },
+  defaultOptions: [],
+  create(context) {
+    const [style] = context.options;
+
+    if (style === 'fields') {
+      return {
+        MethodDefinition(node: TSESTree.MethodDefinition): void {
+          if (
+            node.kind !== 'get' ||
+            !node.value.body ||
+            !node.value.body.body.length
+          ) {
+            return;
+          }
+
+          const [statement] = node.value.body.body;
+
+          if (statement.type !== AST_NODE_TYPES.ReturnStatement) {
+            return;
+          }
+
+          const { argument } = statement;
+
+          if (!argument || !isSupportedLiteral(argument)) {
+            return;
+          }
+
+          context.report({
+            node: node.key,
+            messageId: 'preferFieldStyle',
+            fix(fixer) {
+              const keyOffset = node.computed ? 1 : 0;
+
+              return [
+                // replace the start bits with the field modifiers
+                fixer.replaceTextRange(
+                  [node.range[0], node.key.range[0] - keyOffset],
+                  printNodeModifiers(node, 'readonly'),
+                ),
+                // replace the middle bits with the assignment
+                fixer.replaceTextRange(
+                  [node.value.range[0], argument.range[0]],
+                  '=',
+                ),
+                // remove the end bits
+                fixer.removeRange([argument.range[1], node.range[1]]),
+              ];
+            },
+          });
+        },
+      };
+    }
+
+    if (style === 'getters') {
+      return {
+        ClassProperty(node: TSESTree.ClassProperty): void {
+          if (!node.readonly) {
+            return;
+          }
+
+          const { value } = node;
+
+          if (!value || !isSupportedLiteral(value)) {
+            return;
+          }
+
+          context.report({
+            node: node.key,
+            messageId: 'preferGetterStyle',
+            fix(fixer) {
+              const keyOffset = node.computed ? 1 : 0;
+
+              return [
+                // replace the start bits with the getter modifiers
+                fixer.replaceTextRange(
+                  [node.range[0], node.key.range[0] - keyOffset],
+                  printNodeModifiers(node, 'get'),
+                ),
+                // replace the middle bits with the start of the getter
+                fixer.replaceTextRange(
+                  [node.key.range[1] + keyOffset, value.range[0]],
+                  '(){return ',
+                ),
+                // replace the end bits with the end of the getter
+                fixer.replaceTextRange([value.range[1], node.range[1]], '}'),
```

Comment:
I don't know how interactions between rules work, but how would this interact with the [block-spacing](https://eslint.org/docs/rules/block-spacing) eslint rule? Same question about the [semi](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md) rule for the return statement.

Sorry @bradzacher, I just wasn't clear from your comment if these other rules would fix this if they are set based on your remark about "having to edit it afterward".

Reply:
The docs explain this better than I can:

https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes

> After applying fixes, ESLint will run all of the enabled rules again on the fixed code, potentially applying more fixes. This process will repeat up to 10 times, or until no more fixable problems are found. Afterwards, any remaining problems will be reported as usual.

> Best practices for fixes:
> \4. Since all rules are run again after the initial round of fixes is applied, it's not necessary for a rule to check whether the code style of a fix will cause errors to be reported by another rule.

One exception to this is fixers applied via an IDE - fixing a single error via the IDE will run the fixer exactly once.

Which is why when applying larger fixes like this, I still like to try and go with what I believe to be the "most common" styling, so there's less chance of the fix causing a lint error, and so people are less likely to want to fix it up manually afterward.

Reply:
> One exception to this is fixers applied via an IDE - fixing a single error via the IDE will run the fixer exactly once.

That's a good point - I'm not particularly fussed, and do prefer the nicer output in the tests anyway, so +1